### PR TITLE
resolves #1055 upgrade C4 PlantUML definition files to 2.4.0

### LINKED ISSUE #1055: Upgrade C4 PlantUML to version 2.4.0
[C4 PlantUML 2.4.0](https://github.com/plantuml-stdlib/C4-PlantUML/releases/tag/v2.4.0) was released 2021-10-29.

Implemented enhancements:
* https://github.com/plantuml-stdlib/C4-PlantUML/pull/177
* https://github.com/plantuml-stdlib/C4-PlantUML/pull/173

Current version of C4 PlantUML in Kroki is 2.3.0:

https://github.com/yuzutech/kroki/blob/44f30f9ecb0e65d676318e23083e871564552699/server/src/main/resources/c4.puml#L2

### PR DESCRIPTION
**Pull request by mikaello**
Copied from: https://github.com/plantuml-stdlib/C4-PlantUML/tree/v2.4.0

resolves #1055

### CODE REVIEW

File: server/src/main/resources/c4.puml [hunk 1]
```diff
@@ -1,5 +1,18 @@
 ' C4-PlantUML
-' https://github.com/plantuml-stdlib/C4-PlantUML/tree/v2.3.0
+' https://github.com/plantuml-stdlib/C4-PlantUML/tree/v2.4.0
```
Comment (mikaello) [inline server/src/main/resources/c4.puml:2]: This line was not part of https://github.com/plantuml-stdlib/C4-PlantUML/blob/v2.4.0/C4.puml , but added for consistency.

File: server/src/main/resources/c4_component.puml [hunk 2]
```diff
@@ -1,3 +1,10 @@
+' convert it with additional command line argument -DRELATIVE_INCLUDE="." to use locally
+!if %variable_exists("RELATIVE_INCLUDE")
+  !include %get_variable_value("RELATIVE_INCLUDE")/C4_Container.puml
+!else
+  !include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
+!endif
+
```
Comment (Mogztter) [inline server/src/main/resources/c4_component.puml:7]: We should not include this part, could you please remove it?

File: server/src/main/resources/c4_container.puml [hunk 3]
```diff
@@ -1,3 +1,10 @@
+' convert it with additional command line argument -DRELATIVE_INCLUDE="." to use locally
+!if %variable_exists("RELATIVE_INCLUDE")
+  !include %get_variable_value("RELATIVE_INCLUDE")/C4_Context.puml
+!else
+  !include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Context.puml
+!endif
+
```
Comment (Mogztter) [inline server/src/main/resources/c4_container.puml:7]: ditto

File: server/src/main/resources/c4_context.puml [hunk 4]
```diff
@@ -1,3 +1,10 @@
+' convert it with additional command line argument -DRELATIVE_INCLUDE="." to use locally
+!if %variable_exists("RELATIVE_INCLUDE")
+  !include %get_variable_value("RELATIVE_INCLUDE")/C4.puml
+!else
+  !include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4.puml
+!endif
+
```
Comment (Mogztter) [inline server/src/main/resources/c4_context.puml:7]: ditto

File: server/src/main/resources/c4_deployment.puml [hunk 5]
```diff
@@ -1,3 +1,10 @@
+' convert it with additional command line argument -DRELATIVE_INCLUDE="." to use locally
+!if %variable_exists("RELATIVE_INCLUDE")
+  !include %get_variable_value("RELATIVE_INCLUDE")/C4_Container.puml
+!else
+  !include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Container.puml
+!endif
+
```
Comment (Mogztter) [inline server/src/main/resources/c4_deployment.puml:7]: ditto

File: server/src/main/resources/c4_dynamic.puml [hunk 6]
```diff
@@ -1,3 +1,10 @@
+' convert it with additional command line argument -DRELATIVE_INCLUDE="." to use locally
+!if %variable_exists("RELATIVE_INCLUDE")
+  !include %get_variable_value("RELATIVE_INCLUDE")/C4_Component.puml
+!else
+  !include https://raw.githubusercontent.com/plantuml-stdlib/C4-PlantUML/master/C4_Component.puml
+!endif
+
```
Comment (Mogztter) [inline server/src/main/resources/c4_dynamic.puml:7]: ditto

**Review by Mogztter — CHANGES_REQUESTED** [review]
Since we are loading C4 files from the classpath we need to remove the logic that includes dependent files using the `!include` directive.

Apart from that 👍🏻

File: server/src/main/resources/c4_component.puml (hunk 2 above)
Comment (mikaello) [reply server/src/main/resources/c4_component.puml]: Removed.

File: server/src/main/resources/c4_container.puml (hunk 3 above)
Comment (mikaello) [reply server/src/main/resources/c4_container.puml]: Removed.

File: server/src/main/resources/c4_context.puml (hunk 4 above)
Comment (mikaello) [reply server/src/main/resources/c4_context.puml]: Removed.

File: server/src/main/resources/c4_deployment.puml (hunk 5 above)
Comment (mikaello) [reply server/src/main/resources/c4_deployment.puml]: Removed.

File: server/src/main/resources/c4_dynamic.puml (hunk 6 above)
Comment (mikaello) [reply server/src/main/resources/c4_dynamic.puml]: Removed.

**Comment by Mogztter** [comment]
Thank you for your contribution 🙌🏻

**Pull request merged**